src/libstd/error.rs

```diff
@@ -71,6 +74,14 @@ pub trait Error: Debug + Display {
     /// The lower-level cause of this error, if any.
     #[stable(feature = "rust1", since = "1.0.0")]
     fn cause(&self) -> Option<&Error> { None }
+
+    /// Get the `TypeId` of `self`
+    #[doc(hidden)]
+    #[unstable(feature = "core",
+               reason = "unclear whether to commit to this public implementation detail")]
+    fn type_id(&self) -> TypeId where Self: 'static {
```

Comment:
@reem Ah yes, I think at some point in the iterations on this topic I confused myself into thinking this would hurt more than it does: you can still have non-`'static` generic error types, they will just fail to be `Error` for the time being. That seems plausible, and I agree that not stabilizing `Reflect` is a good idea.